e2e-tests/upgrade/run

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+
+set -o errexit
+set -o xtrace
+
+test_dir=$(realpath $(dirname $0))
+. ${test_dir}/../functions
+
+CONTAINERS=(pxc proxysql)
+CLUSTER='upgrade'
+IMAGE_TO_UPDATE=${IMAGE}
+IMAGE_PXC_TO_UPDATE=${IMAGE_PXC}
+IMAGE_PROXY_TO_UPDATE=${IMAGE_PROXY}
+IMAGE_BACKUP_TO_UPDATE=${IMAGE_BACKUP}
+IMAGE='percona/percona-xtradb-cluster-operator:1.0.0'
+IMAGE_PXC='percona/percona-xtradb-cluster-operator:1.0.0-pxc'
+IMAGE_PROXY='percona/percona-xtradb-cluster-operator:1.0.0-proxysql'
+IMAGE_BACKUP='percona/percona-xtradb-cluster-operator:1.0.0-backup'
+OPERATOR_NAME='percona-xtradb-cluster-operator'
+
+
+wait_cluster_consistency() {
+    cluster_name=$1
+    cluster_size=$2
+    until [[ "$(kubectl get pxc "${cluster_name}" -o jsonpath='{.status.state}')" == "ready" \
+             && "$(kubectl get pxc "${cluster_name}" -o jsonpath='{.status.pxc.ready}')" == "${cluster_size}" \
+             && "$(kubectl get pxc "${CLUSTER}" -o jsonpath='{.status.proxysql.ready}')" == "${cluster_size}"  ]]; do
+        echo 'waiting for cluster readyness'
+        sleep 20
+    done
+}
+
+check_pxc_livenes() {
+    cluster_name=$1
+    cluster_size=$2
+
+    wait_cluster_consistency "${cluster_name}" "${cluster_size}"
+    sleep 20
+    wait_cluster_consistency "${cluster_name}" "${cluster_size}"
+    sleep 20
+    wait_cluster_consistency "${cluster_name}" "${cluster_size}"
+    sleep 20
+    wait_cluster_consistency "${cluster_name}" "${cluster_size}"
+    
+    wait_for_running "${cluster_name}-pxc" "${cluster_size}"
+
+    for i in $(seq 0 $(($cluster_size-1))); do
+        compare_mysql_cmd "select-1" "SELECT * from myApp.myApp;" "-h ${cluster_name}-pxc-$i.${cluster_name}-pxc -uroot -proot_password"
+    done
+}
+
+compare_generation() {
+    step=$1
+    for name in ${CONTAINERS[@]}; do
+        kubectl get statefulset "${CLUSTER}-${name}" -o jsonpath='{.metadata.generation}' > ${tmp_dir}/"${CLUSTER}-${name}".gen
+        diff "${test_dir}/compare/${step}.gen" "${tmp_dir}/${CLUSTER}-${name}.gen"
+    done
+}
+
+main() {
+    create_namespace $namespace
+    deploy_operator
+
+    spinup_pxc "${CLUSTER}" "${test_dir}/conf/${CLUSTER}.yml"
+
+    compare_generation "before_upgrade"
+
+    kubectl patch deployment "${OPERATOR_NAME}" \
+        -p'{"spec":{"template":{"spec":{"containers":[{"name":"'"${OPERATOR_NAME}"'","image":"'"${IMAGE_TO_UPDATE}"'"}]}}}}'
+    kubectl rollout status deployment/"${OPERATOR_NAME}"
+
+    echo 'Waiting for operator upgrade'
+    until [[ $(kubectl get pods --selector=name=${OPERATOR_NAME} \
```

Comment:
```suggestion
    wait_pod $(get_operator_pod)
```

Reply:
we need to exclude the pod in terminating status on the image check line. Either we wait here or  we have figure out something else